lib/ts/recipeImplementation.ts

```diff
@@ -303,11 +303,7 @@ export default function RecipeImplementation(recipeImplInput: {
                         domain = urlObj.port === "" ? domain : domain + ":" + urlObj.port;
                     }
                 }
-                if (sessionTokenBackendDomain.startsWith(".")) {
```

Comment:
also fix the normaliseSessionScopeOrThrowError functino to remove the redundant check, similar to how it's done in the node sdk

Reply:
Done